massa-protocol-worker/src/handlers/peer_handler/tester.rs

```diff
@@ -428,11 +428,23 @@ impl Tester {
                     default(Duration::from_secs(2)) => {
                         // If no message in 2 seconds they will test a peer that hasn't been tested for long time
 
-                        let Some(listener) = db.read().get_oldest_peer(protocol_config.test_oldest_peer_cooldown.into()) else {
+                        let res = {
+                            let mut peers_in_test = peers_in_test.write();
+
+                            let res = db.read().get_oldest_peer(
+                                protocol_config.test_oldest_peer_cooldown.into(),
+                                &peers_in_test,
+                            );
+                            if let Some(ref listener) = res {
+                                peers_in_test.insert(*listener);
```

Comment:
is peers_in_test cleared somewhere ? looks like it grows forever

Reply:
it was deleted after Tester::tcp_handshake()

Reply:
can you make sure there is no return or ? etc... before that makes us not delete it ?

Reply:
Added a forgotten case (only happen on a single address, so doesn't expand indefinitly) but as we already did a pass in the past (because of a related issue), we delete it before any return / continue

Reply:
perfect !